conan/tools/cmake/toolchain/blocks.py

```diff
@@ -559,6 +559,31 @@ def context(self):
         }
 
 
+class PkgConfigBlock(Block):
+    template = textwrap.dedent("""
+        {% if pkg_config %}
+        set(PKG_CONFIG_EXECUTABLE {{ pkg_config }} CACHE FILEPATH "pkg-config executable")
+        {% endif %}
+        {% if pkg_config_path %}
+        set(ENV{PKG_CONFIG_PATH} "{{ pkg_config_path }}")
+        {% endif %}
+        """)
+
+    def context(self):
+        pkg_config = self._conanfile.conf.get("tools.gnu:pkg_config", check_type=str)
+        if pkg_config:
+            pkg_config = pkg_config.replace("\\", "/")
+        pkg_config_path = self._conanfile.generators_folder
+        if pkg_config_path:
+            # Check if user has already defined their own PKG_CONFIG_PATH variable
+            user_pkg_config_path = os.getenv("PKG_CONFIG_PATH")
+            if user_pkg_config_path:
+                # Prepending the Conan one
+                pkg_config_path = pkg_config_path + os.pathsep + user_pkg_config_path
```

Comment:
Should I use this approach from `EnvVars`?

```python
        self._subsystem = deduce_subsystem(conanfile, scope)

    @property
    def _pathsep(self):
        return ":" if self._subsystem != WINDOWS else ";"
```

Reply:
Yes, I think the correct way to do this is to deduce the subsystem and then aplying the path separator according to that

Reply:
The question is: should I assume that `scope == "build"`? Here, we don't have any scope defined.